Project 2/k-NNAlgorithms/src/FireData.java

```diff
@@ -0,0 +1,21 @@
+import java.io.File;
+
+public class FireData extends Data {
+
+    FireData(File inputFileName) throws Exception {
+        fileTo2dStringArrayList(inputFileName);
+        preProcess();
+        normalizeData();
+        bucketize();
+    }
+
+    private void preProcess() {
+        for (int i = 0; i < fullSet.size(); i++) {
+            //removes the data representing the months
+            //and days of the week as these will
+            //not be valid data points using euclidean distance
+            fullSet.get(i).remove(2);
+            fullSet.get(i).remove(2);
```

Comment:
It's because the indices are shifting when things are removed from the arraylists, this is removing column 3 data, then column 4 data respectively

Reply:
Figured this might be the case. Might be able to do something like: 
`list.subList(indexStart, indexEnd).clear()`
to make this a little more clean and give those indexes meaningful names.